site-docker-tripleo.yml.sample

```diff
@@ -0,0 +1,105 @@
+---
+# Defines deployment design and assigns role to server groups
+
+- hosts:
+  - mons
+  - agents
+  - osds
+  - mdss
+  - rgws
+  - nfss
+  - restapis
+  - rbdmirrors
+  - clients
+  - iscsigws
+  - mgrs
+
+  gather_facts: false
+
+  tasks:
+    - name: gather and delegate facts
+      setup:
+      delegate_to: "{{ item }}"
+      delegate_facts: True
+      with_items: "{{ groups['all'] }}"
+
+- hosts: mons
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-mon
+  serial: 1 # MUST be '1' WHEN DEPLOYING MONITORS ON DOCKER CONTAINERS
+
+- hosts: osds
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-osd
+
+- hosts: mdss
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-mds
+
+- hosts: rgws
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-rgw
+
+- hosts: nfss
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-nfs
+
+- hosts: rbd_mirrors
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-rbd-mirror
+
+- hosts: restapis
+  become: True
+  gather_facts: false
+  roles:
+    - ceph-defaults
+    - ceph-docker-common
+    - ceph-config
+    - ceph-restapi
+
+- hosts: mgrs
+  become: True
+  gather_facts: false
```

Comment:
My testing of this with a client-only deployment failed with 'ansible_hostname' is undefined. When I set gather_facts to true it seemed to correctly configure the ceph client. 

2017-09-08 17:29:50,269 p=9448 u=mistral |  fatal: [192.168.24.22]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'ansible_hostname' is undefined\n\nThe error appears to have been in '/usr/share/ceph-ansible/roles/ceph-defaults/tasks/facts.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- set_fact:\n  ^ here\n"}

Reply:
@fultonj what version of ansible are you using? This playbook will require ``ansible>=2.3.1`` and this error kind of looks like one we see when ``delegate_facts: True`` is used with ansible versions older than ``2.3.1``.

Reply:
@andrewschoen TripleO uses Ansible 2.3.0 so I'm worried this will not work with it. I opened an issue on this here: https://github.com/ceph/ceph-ansible/issues/1876

Reply:
Then I think we can solve #1876 with this PR. Let me update it.